src/overlays/actors/ovl_Boss_03/z_boss_03.c

```diff
@@ -4,31 +4,259 @@
  * Description: Gyorg
  */
 
+/**
+ * Some notes:
+ *
+ * Global flags:
+ * - gSaveContext.eventInf[5] & 0x40: Enabled when Gyorg's intro cutscene has been watched
+ * - gSaveContext.weekEventReg[55] & 0x80: Checked to know if Gyorg has been defeated
+ *
+ * Seaweed:
+ * - Refers to the seaweed at the bottom of the stage where Player fights Gyorg
+ * - The default Gyorg actor will spawn 5 other Gyorg instances using the parameter GYORG_PARAM_SEAWEED to spawn them
+ * - Seaweed can "interact" only with the main Gyorg instance and Player
+ *
+ * This actor makes some heavy assumptions about the positions of the room where it is spawned
+ *
+ * Cutscenes:
+ * - There are 3 cutscenes:
+ *   - IntroCutscene: The cs which is played when Player falls into the main room. It also shows Gyorg's titlecard
+ *   - SpawnSmallFishesCutscene: The short cs which is played when Gyorg is spawning the small fishes (EnTanron3). This
+ * is triggered when Gyorg reaches half of its life.
+ *   - DeathCutscene: Played when Gyorg dies. Showing it splashing and becoming smaller each time until it disappears
+ * - This actor mainly handles the 3 cutscenes it has manually (instead of relying on existing systems for it)
+ *
+ * Main behaviour:
+ * - Gyorg has two branches on Gyorg's behaviour depending on Player's state:
+ *   - If Player is standing on the main platform, then Gyorg follows the PrepareCharge -> Charge -> JumpOverPlatform
+ * branch
+ *   - Otherwise, Gyorg follows the ChasePlayer -> CatchPlayer -> ChewPlayer
+ * - The main actionFunc which decides which branch should be taken is func_809E34B8
+ * - Most of the actions of those two branches are constantly checking for the WORK_TIMER_CURRENT_ACTION timer. If it
+ * runs out, then the behaviour resets back to func_809E34B8
+ * - Either branch behaviour can be interrupted at any time by a hit from Player
+ *   - Being hit once makes Gyorg Stunned
+ *   - When Gyorg is Stunned, it is vulnerable to be Damaged by Player
+ *   - When Gyorg's health drops below half, it spawns the small fishes
+ *
+ * The collision logic is handled manually
+ */
+
+#include "prevent_bss_reordering.h"
 #include "z_boss_03.h"
+#include "overlays/actors/ovl_Door_Warp1/z_door_warp1.h"
+#include "overlays/actors/ovl_En_Water_Effect/z_en_water_effect.h"
+#include "objects/gameplay_keep/gameplay_keep.h"
+#include "objects/object_water_effect/object_water_effect.h"
 
 #define FLAGS (ACTOR_FLAG_1 | ACTOR_FLAG_4 | ACTOR_FLAG_10 | ACTOR_FLAG_20)
 
 #define THIS ((Boss03*)thisx)
 
+#define WORK_TIMER_UNK0_A 0 // used in func_809E34B8
+#define WORK_TIMER_CURRENT_ACTION 0
+#define WORK_TIMER_UNK0_C 0 // used in DeathCutscene
+
+#define WORK_TIMER_UNK1_A 1 // used in func_809E34B8
+#define WORK_TIMER_UNK1_B 1 // used in PrepareCharge
+
+#define WORK_TIMER_UNK2_A 2 // used in func_809E34B8
+#define WORK_TIMER_STUNNED 2
+
+// The Y position of the top of the platform.
```

Comment:
Now the lack of full stops in the top comment is at variance with stuff like this. I'd rather have full stops everywhere, really.